papis/commands/add.py

```diff
@@ -305,36 +304,39 @@ def run(paths: List[str],
     tmp_document = papis.document.Document(temp_dir)
 
     if base_path is None:
-        base_path = pathlib.Path(papis.config.get_lib_dirs()[0]).expanduser()
+        base_path = os.path.expanduser(papis.config.get_lib_dirs()[0])
     out_folder_path = base_path
 
     if subfolder:
-        out_folder_path /= pathlib.Path(subfolder)
+        out_folder_path = os.path.join(out_folder_path, subfolder)
 
     if not folder_name:
         out_folder_name = get_hash_folder(data, in_documents_paths)
-        out_folder_path = out_folder_path / out_folder_name
+        out_folder_path = os.path.join(out_folder_path, out_folder_name)
         logger.info("Got an automatic folder name")
     else:
         temp_doc = papis.document.Document(data=data)
-        temp_path = out_folder_path / pathlib.Path(folder_name)
-        components: List[str] = []
-        assert temp_path.is_relative_to(out_folder_path)
-        while temp_path != out_folder_path and temp_path.is_relative_to(out_folder_path):
-            path_component = temp_path.name
-            component_formatted = papis.format.format(path_component, temp_doc)
-            component_cleaned = papis.utils.clean_document_name(component_formatted)
+        temp_path = os.path.join(out_folder_path, folder_name)
+        components = []     # type: List[str]
+        while (
+                temp_path != out_folder_path
+                and papis.utils.is_relative_to(temp_path, out_folder_path)):
+            path_component = os.path.basename(temp_path)
+
+            component_cleaned = papis.utils.clean_document_name(
+                papis.format.format(path_component, temp_doc))
             components.insert(0, component_cleaned)
```

Comment:
This whole chunk was ported back to using `os.path` instead of `pathlib`. There were a bunch of failures on `macos` related to `pathlib` that I haven't been able to diagnose
https://github.com/papis/papis/runs/8135558982?check_suite_focus=true#step:5:134

Probably work for some other time to port everything to using the nicer `pathlib` interfaces.

Reply:
I am not sure pathlib supports python3.5, at some point maybe we'll have to drop python3.5, but yes, it's good to have the os.path back